src/lib.rs

```diff
@@ -284,20 +282,30 @@ functions:
     fn xkb_compose_state_get_one_sym(*mut xkb_compose_state) -> xkb_keysym_t,
 );
 
-lazy_static!(
-    pub static ref XKBCOMMON_OPTION: Option<XkbCommon> = {
-        open_with_sonames(&["libxkbcommon.so", "libxkbcommon.so.0"], None, |name| unsafe { XkbCommon::open(name) })
-    };
-    pub static ref XKBCOMMON_HANDLE: &'static XkbCommon = {
-        XKBCOMMON_OPTION.as_ref().expect("Library libxkbcommon.so could not be loaded.")
-    };
-    pub static ref XKBCOMMON_COMPOSE_OPTION: Option<XkbCommonCompose> = {
-        open_with_sonames(&["libxkbcommon.so", "libxkbcommon.so.0"], Some("compose"), |name| unsafe { XkbCommonCompose::open(name) })
-    };
-    pub static ref XKBCOMMON_COMPOSE_HANDLE: &'static XkbCommonCompose = {
-        XKBCOMMON_COMPOSE_OPTION.as_ref().expect("Could not load compose module from libxkbcommon.so.")
-    };
-);
+pub static XKBCOMMON_OPTION: Lazy<Option<XkbCommon>> = Lazy::new(|| {
+    open_with_sonames(
+        &["libxkbcommon.so", "libxkbcommon.so.0"],
+        None,
+        |name| unsafe { XkbCommon::open(name) },
+    )
+});
+pub static XKBCOMMON_HANDLE: Lazy<&'static XkbCommon> = Lazy::new(|| {
```

Comment:
I think it would be better to not expose the `Lazy` type, since [`OnceLock`](https://doc.rust-lang.org/std/sync/struct.OnceLock.html) is now in `std` and making the `once_cell` crate a public dependency [is fragile](https://github.com/matklad/once_cell/issues/201). I'd prefer an approach like this:

```rust
pub fn xkbcommon_handle() -> &'static XkbCommon {
    static XKBCOMMON_HANDLE: OnceCell<XkbCommon> = OnceCell::new();
    XKBCOMMON_HANDLE.get_or_init(|| /* ... */)
}
```

This would need elinorbgr/dlib#21 to be released first.

Reply:
Yeah, we can open follow up for that, I wouldn't release this crate for a while.

The crate on its own is weird and its scope is sort of limited, as in for winit only.